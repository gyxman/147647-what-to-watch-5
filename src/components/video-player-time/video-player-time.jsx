import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {setCurrentMovieDuration, setCurrentMovieTime} from "../../store/action";

const VideoPlayerTime = (props) => {
  const {getVideo, onLoaded, duration, currentTime, setDuration, setTime} = props;

  setTimeout(() => {
    const video = getVideo().current;

    video.oncanplaythrough = (e) => {
      onLoaded();
      setDuration(e.target.duration);
    };

    video.addEventListener(`progress`, function (e) {
      const time = e.target.currentTime;
      if (Math.floor(time) > currentTime) {
        setTime(time);
      }
    });
  }, 0);


  return <React.Fragment>
    <div className="player__time">
      <progress className="player__progress" value={duration ? currentTime / duration * 100 : 0} max="100"/>
      <div className="player__toggler" style={{left: `${currentTime / duration * 100}%`}}>Toggler</div>
    </div>
    <div className="player__time-value">1:30:29</div>
  </React.Fragment>;
};

VideoPlayerTime.propTypes = {
  getVideo: PropTypes.func.isRequired,
  onLoaded: PropTypes.func.isRequired,
  duration: PropTypes.number.isRequired,
  currentTime: PropTypes.number.isRequired,
  setDuration: PropTypes.func.isRequired,
  setTime: PropTypes.func.isRequired,
};

const mapStateToProps = ({PROCESS}) => ({
  duration: PROCESS.currentMovieDuration,
  currentTime: PROCESS.currentMovieTime
});

const mapDispatchToProps = (dispatch) => ({
  setDuration(duration) {
    dispatch(setCurrentMovieDuration(duration));
  },
  setTime(time) {
    dispatch(setCurrentMovieTime(time));
  },
});

export {VideoPlayerTime};
export default connect(mapStateToProps, mapDispatchToProps)(VideoPlayerTime);
