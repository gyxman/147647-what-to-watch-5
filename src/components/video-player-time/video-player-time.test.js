import React from "react";
import renderer from "react-test-renderer";
import {VideoPlayerTime} from "./video-player-time";

const noop = () => {};

it(`VideoPlayerTime is rendered correctly`, () => {
  const tree = renderer.create(
      <VideoPlayerTime onLoaded={noop} currentTime={0} duration={1} getVideo={noop} setDuration={noop} setTime={noop} />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
