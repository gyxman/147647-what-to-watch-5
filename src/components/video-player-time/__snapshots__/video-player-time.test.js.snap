// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPlayerTime is rendered correctly 1`] = `
Array [
  <div
    className="player__time"
  >
    <progress
      className="player__progress"
      max="100"
      value={0}
    />
    <div
      className="player__toggler"
      style={
        Object {
          "left": "0%",
        }
      }
    >
      Toggler
    </div>
  </div>,
  <div
    className="player__time-value"
  >
    01
  </div>,
]
`;
